### DISCUSSION TITLE
IPv4 links

### DISCUSSION THREAD

**Discussion by ge2301** [Tyrrrz/YoutubeExplode#747] [General]
We are using YoutubeExlore as plugin in a HTPC software, that e.g. can player trailers. Unfortunately created IPv6 links are not playable. Is it possible to enforce IPv4 links to the video file?
Here a similar issue was resolved: https://github.com/ytdl-org/youtube-dl/issues/3626

**Comment by Tyrrrz** [comment]
Where do you see IPv6 links? Can you give an example?

**Reply by ge2301** [reply]
YoutubeExplore preferrably create IPv6 links, which result in: 
Access to rr2---sn-h0jeener.googlevideo.com was denied
You are not authorized to access this page.
HTTP ERROR 403

The bug in not in your code, but youtube itself. The only way to solve it is to force IPv4 links.

Example Link: https://rr2---sn-h0jeener.googlevideo.com/videoplayback?expire=1677427940&ei=hDD7Y_mIBI2f-gacn6HwAw&ip=2003%3Afa%3Aef11%3A1300%3A9c90%3Aecb4%3A984f%3Ab29c&id=o-AB2_01PbEuCtSP5robtxbbEnwuLLiKx95MupP6kF4nlm&itag=22&source=youtube&requiressl=yes&mh=PM&mm=31%2C29&mn=sn-h0jeener%2Csn-h0jelne6&ms=au%2Crdu&mv=m&mvi=2&pl=37&initcwndbps=1388750&spc=H3gIhmlF_QTor2mN4QbmkXKvaP1L4m094o_z73BDUUKR0_ZftA&vprv=1&mime=video%2Fmp4&cnr=14&ratebypass=yes&dur=162.005&lmt=1666311152247472&mt=1677405901&fvip=3&fexp=24007246&c=ANDROID&txp=5532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Cmime%2Ccnr%2Cratebypass%2Cdur%2Clmt&sig=AOq0QJ8wRgIhALAIzkJiG2he1MM_oUw3bzQsoQGVwzEiaeaQOgHAxfb-AiEAjgMywKbRlCkt_zc2p1zTKghQriMAd6OXV5wTpbGe3v4%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgQ80Sc4WfvDHVJNcK2rLtduVY23bNR4E79sfJjsH5ED0CIQD7hejI95ZQiipXL9M-xYAMSIn

**Reply by Tyrrrz** [reply]
Ah, you mean the `ip` parameter corresponds to an IPV6 address? I haven't seen this before and I'm not aware of how to force it to provide the URL in a different format.

**Reply by ge2301** [reply]
Yes, that‘s what I mean. Maybe the link in my initial post helps, because they are also open source and tackled exactly this issue.

**Reply by Tyrrrz** [reply]
That issue is unrelated to this
